File: mcspring-build/mcspring-plugin-manager/src/main/java/in/kyle/mcspring/manager/commands/CommandClassLoader.java
package in.kyle.mcspring.manager.commands;

import org.springframework.stereotype.Component;

import in.kyle.mcspring.command.Command;
import in.kyle.mcspring.subcommands.PluginCommand;

@Component
class CommandClassLoader {
    
    @Command(value = "classloader",
             description = "Show ClassLoader information for a specific class",
             usage = "/classloader <class>")
    void classLoader(PluginCommand command) {
        command.withString();
        command.then(this::executeClassLoader);
        command.otherwise("Usage: /classloader <class>");
    }
    
    String executeClassLoader(String clazz) {
        try {
            Class<?> aClass = Class.forName(clazz);
            String classLoader = aClass.getClassLoader().toString();
            String protectionDomain =
                    aClass.getProtectionDomain().getCodeSource().getLocation().toString();
            return String.format("ClassLoader: %s\nDomain: %s", classLoader, protectionDomain);
        } catch (ClassNotFoundException e) {
            return String.format("Class %s not found", clazz);
        }
    }
}

File: mcspring-build/mcspring-plugin-manager/src/main/java/in/kyle/mcspring/manager/commands/CommandClassLoader.kt
package `in`.kyle.mcspring.manager.commands

import `in`.kyle.mcspring.command.Command
import `in`.kyle.mcspring.subcommands.plugincommand.PluginCommandImpl
import org.springframework.stereotype.Component

@Component
internal class CommandClassLoader {

    @Command(value = "classloader", description = "Show ClassLoader information for a specific class", usage = "/classloader <class>")
    fun classLoader(command: PluginCommandImpl) {
        command.withString();
        command.then(::executeClassLoader);
        command.otherwise("Usage: /classloader <class>")
    }

    private fun executeClassLoader(clazz: String): String {
        val aClass = Class.forName(clazz)
        val classLoader = aClass.classLoader.toString()
        val protectionDomain = aClass.protectionDomain.codeSource.location.toString()
        return """
            ClassLoader: $classLoader
            Domain: $protectionDomain
        """.trimIndent()
    }
}
